backend/modules/model_gateway/model_gateway.py

```diff
@@ -1,85 +1,119 @@
+                        model_name = f"{list.provider_name}/{model_id}"
+                        logger.info(f"Adding model {model_name} to the model gateway.")
+                        self.modelsToProviderMap[model_name] = list
+                    else:
+                        # Avoid adding the model to the map if the API key is not set
+                        logger.warning(
+                            f"Environment variable {list.api_key_env_var} not set."
+                        )
+
     def get_embedding_models(self) -> List[ModelConfig]:
         models: List[ModelConfig] = []
         for list in self.config:
-            for model_id in list.embedding_model_ids:
-                models.append(ModelConfig(
-                    name=f"{list.provider_name}/{model_id}", 
-                    type=ModelType.embedding
-                ).dict())
-                              
+            if list.api_key_env_var and os.environ.get(list.api_key_env_var):
+                # Register only those models whose API key is set
+                for model_id in list.embedding_model_ids:
+                    models.append(
+                        ModelConfig(
+                            name=f"{list.provider_name}/{model_id}",
+                            type=ModelType.embedding,
+                        ).dict()
+                    )
+
         return models
-    
+
     def get_llm_models(self) -> List[ModelConfig]:
         models: List[ModelConfig] = []
         for list in self.config:
-            for model_id in list.llm_model_ids:
-                models.append(ModelConfig(
-                    name=f"{list.provider_name}/{model_id}", 
-                    type=ModelType.chat
-                ).dict())
-                              
+            if list.api_key_env_var and os.environ.get(list.api_key_env_var):
+                # Register only those models whose API key is set
+                for model_id in list.llm_model_ids:
+                    models.append(
+                        ModelConfig(
+                            name=f"{list.provider_name}/{model_id}", type=ModelType.chat
+                        ).dict()
+                    )
+
         return models
-    
+
     def get_embedder_from_model_config(self, model_name: str) -> Embeddings:
         if model_name not in self.modelsToProviderMap:
             raise ValueError(f"Model {model_name} not registered in the model gateway.")
-        model_provider_config: ModelProviderConfig = self.modelsToProviderMap[model_name]
+        model_provider_config: ModelProviderConfig = self.modelsToProviderMap[
+            model_name
+        ]
         if not model_provider_config.api_key_env_var:
             api_key = None
         else:
-            api_key = os.environ.get(model_provider_config.api_key_env_var, '')
+            api_key = os.environ.get(model_provider_config.api_key_env_var, "")
         model_id = "/".join(model_name.split("/")[1:])
```

Comment:
We should be clear what is considered id and what is considered name
just from reading I would assume id is the fully qualified name but here is seems reverse